models.py

```diff
@@ -326,23 +299,59 @@ def __str__(self):
             self.proofreader.full_name(),
         )
 
-    def cancel(self):
+    def cancel(self, request):
         self.cancelled = True
         self.completed = timezone.now()
         self.save()
 
-    def reset(self):
+        kwargs = {
+            'assignment': self,
+            'request': request,
+            'event_type': 'cancelled',
+        }
+
+        events_logic.Events.raise_event(
+            plugin_settings.ON_PROOFREADER_ASSIGN_CANCELLED,
+            task_object=self.round.article,
+            **kwargs,
+        )
+
+    def reset(self, request):
         self.cancelled = False
         self.completed = None
         self.accepted = None
         self.save()
 
-    def complete(self):
+        kwargs = {
+            'assignment': self,
+            'request': request,
+            'event_type': 'reset',
+        }
+
+        events_logic.Events.raise_event(
+            plugin_settings.ON_PROOFREADER_ASSIGN_RESET,
+            task_object=self.round.article,
+            **kwargs,
+        )
+
+    def complete(self, request):
```

Comment:
I don't think the models should depend on the request, since it locks the models from being used outside of the context of an HTTP request. Instead it should either be an optional argument or retrieved from the global context (ideally we would do this from the event framework instead).